src/main/java/com/icthh/xm/uaa/security/DomainJwtAccessTokenConverter.java

```diff
@@ -70,11 +73,12 @@ private void enrichCustomDetails(OAuth2Authentication authentication, Map<String
             } else {
                 details.put(AUTH_LOGINS_KEY, userDetails.getLogins());
                 details.put(AUTH_ROLE_KEY, getOptionalRoleKey(userDetails.getAuthorities()));
+                details.putIfAbsent(AUTH_ADDITIONAL_DETAILS, new HashMap<>());
+                Map<String, Object> additionalDetails = (Map<String, Object>) details.get(AUTH_ADDITIONAL_DETAILS);
                 if (isNotEmpty(userDetails.getAdditionalDetails())) {
-                    details.putIfAbsent(AUTH_ADDITIONAL_DETAILS, new HashMap<>());
-                    Map<String, Object> additionalDetails = (Map<String, Object>) details.get(AUTH_ADDITIONAL_DETAILS);
                     additionalDetails.putAll(userDetails.getAdditionalDetails());
                 }
+                additionalDetails.put(MULTI_ROLE_ENABLED, tenantPropertiesService.getTenantProps().isMultiRoleEnabled());
```

Comment:
why do you permanently add MULTI_ROLE_ENABLED flag to the token even if the feature is turned off? 
Suggest adding it only for the case when feature is ON. Otherwise, invoking party should interprete flag absence as false.